File: randomizer-compiler/src/main/java/com/kostasdrakonakis/randomizer/compiler/ShortProperty.java
package com.kostasdrakonakis.randomizer.compiler;

import java.util.Random;

import javax.lang.model.element.Element;
import javax.lang.model.type.TypeKind;
import javax.lang.model.util.Elements;
import javax.lang.model.util.Types;

class ShortProperty extends RandomProperty {

    ShortProperty(Element element) {
        super(element);
    }

    @Override
    boolean isTypeValid(Elements elements, Types types) {
        return element.asType().getKind().equals(TypeKind.SHORT);
    }

    @Override
    String getRandomValue() {
        Random random = new Random();
        return "" + (short) random.nextInt(1 << 16);
    }
}

File: randomizer-compiler/src/main/java/com/kostasdrakonakis/randomizer/compiler/ShortProperty.kt
package com.kostasdrakonakis.randomizer.compiler

import java.util.Random
import javax.lang.model.element.Element
import javax.lang.model.type.TypeKind
import javax.lang.model.util.Elements
import javax.lang.model.util.Types

internal class ShortProperty(element: Element) : RandomProperty(element) {
    override fun isTypeValid(elements: Elements, types: Types): Boolean {
        return element.asType().kind == TypeKind.SHORT
    }

    override val randomValue: String get() = "${Random().nextInt(1 shl 16).toShort()}"
}